### DISCUSSION TITLE
Is xxhash necessary? backport kitty for Ubuntu 20.04 LTS

### DISCUSSION THREAD

**Discussion by iyongliang** [kovidgoyal/kitty#6582] [Show and tell]
I'm working on Ubuntu 20.04, and I'm interested in the GPU accelerated terminal emulator, but when I want install from release, and found the glibc version was too high for me.

Tried build from source, and found rm -rf linux-package
python3 setup.py linux-package
Package wayland-scanner was not found in the pkg-config search path.
Perhaps you should add the directory containing `wayland-scanner.pc'
to the PKG_CONFIG_PATH environment variable
No package 'wayland-scanner' found
The package wayland-scanner was not found on your system
Disabling building of wayland backend
[17/66] Compiling kittens/transfer/algorithm.c ... done
Compiling kittens/transfer/algorithm.c ...
gcc -MMD -DNDEBUG -DHAS_COPY_FILE_RANGE -Wextra -Wfloat-conversion -Wno-missing-field-initializers -Wall -Wstrict-prototypes -std=c11 -pedantic-errors -Werror -O3 -fwrapv -fstack-protector-strong -pipe -fvisibility=hidden -D_FORTIFY_SOURCE=2 -fPIC -flto -pthread -Ikitty -I/usr/include/python3.8 -I-I/build/xxhash-TCxsRP/xxhash-0.7.3/debian/tmp/usr/include -c kittens/transfer/algorithm.c -o build/rsync-algorithm.c.o
kittens/transfer/algorithm.c: In function ‘xxh64_delete’:
kittens/transfer/algorithm.c:41:37: error: implicit declaration of function ‘XXH3_freeState’; did you mean ‘XXH32_freeState’? [-Wimplicit-function-declaration]
   41 | static void xxh64_delete(void* s) { XXH3_freeState(s); }
      |                                     ^~~~~~~~~~~~~~
      |                                     XXH32_freeState
kittens/transfer/algorithm.c: In function ‘xxh64_reset’:
kittens/transfer/algorithm.c:42:43: error: implicit declaration of function ‘XXH3_64bits_reset’; did you mean ‘XXH64_reset’? [-Wimplicit-function-declaration]
   42 | static bool xxh64_reset(void* s) { return XXH3_64bits_reset(s) == XXH_OK; }
      |                                           ^~~~~~~~~~~~~~~~~
      |                                           XXH64_reset

**Comment by kovidgoyal** [comment]
Yes, it's necessary and you dont need to install anything other than Go to build kitty. Read build.rst

**Reply by iyongliang** [reply]
must upgrade xxhash to 0.8.0, I also try use debian package, same error.

**Comment by kovidgoyal** [comment]
read build.rst from the beginning. You dont need any system libraries
other than compilers. All dependencies are automatically downloaded for
you.

**Reply by iyongliang** [reply]
I'm **readed** build.rst, and download libxxhash-dev from **Ubuntu 20.04LTS OFFICIAL** repository, the version cannot build, and report errot to ahead post. It's ubuntu self package error. And I download xxhash-0.8.1-1 from **22.04** repository and rebuild at 20.04, compile can passed!
It not about kitty, but cannot suitable xxhash below 0.8.0.
if you don't think this a issue, ignore that.

**Reply by kovidgoyal** [reply]
I dont know how many ways to say this. The build script will download and use its own libxxhash. You dont need libxxhash installed in your OS.

**Reply by iyongliang** [reply]
thanks you reply, build.rst have some different with https://sw.kovidgoyal.net/kitty/build/, I got it.
